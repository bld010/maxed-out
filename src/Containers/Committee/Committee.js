import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setCurrentCandidate, setCurrentCommitteeId } from '../../actions/index';
import { Link } from 'react-router-dom';
import Contributions from '../Contributions/Contributions';
import { searchCommitteeById, searchCandidateById, fetchPACContributions } from '../../util/apiCalls';

class Committee extends Component {

  constructor(props) {
    super(props)
    this.state = {
      committee: null,
      candidate: null,
      error: ''
    }
  }



  componentDidMount = async () => {
    
    try {
      let committeeSearchResults = await searchCommitteeById(this.props.committee_id);      
      this.setState({
        committee: committeeSearchResults[0]
      })
      this.props.setCurrentCommitteeId(this.props.committee_id)
      let candidate = await searchCandidateById(this.state.committee.candidate_ids[0])
      this.props.setCurrentCandidate(candidate[0]);
    } catch (err) {
      this.setState({
        error: err.message
      })
    }

  }

  render() {

    return(
      
      
        <section className="Committee">
          {this.state.committee && <p>{this.state.committee.committee_id}</p>}
          {this.props.candidate && <p>{this.props.candidate.name}</p>}


          {this.state.committee && <Contributions type="PAC" committee_id={this.props.committee_id} />}
        </section>
     

    )

  }
}

export const mapStateToProps = state => ({
  candidate: state.candidate
})

export const mapDispatchToProps = dispatch => ({
  setCurrentCandidate: candidate => dispatch(setCurrentCandidate(candidate)),
  setCurrentCommitteeId: (committee_id) => dispatch(setCurrentCommitteeId(committee_id))
})


export default connect(mapStateToProps, mapDispatchToProps)(Committee);

//PropTypes