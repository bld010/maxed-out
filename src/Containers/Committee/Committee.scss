.Committee {

  display: flex;
  flex-direction: column;
  // background: pink;

  & .contributions-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    min-height: 600px;
    height: 70vh;
    padding: 20px;
  }

  & .name-and-info h3 {
    text-align: center;
    font-size: 20px;
  }

  & .name-and-info p {
    font-size: 14px;
    text-align: center;
  }
}