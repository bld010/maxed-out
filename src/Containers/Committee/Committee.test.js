import React from 'react';
import { shallow } from 'enzyme';
import { Committee, mapStateToProps, mapDispatchToProps } from './Committee' 
import {searchCommitteeById,
        fetchIndividualContributions,
        fetchPACContributions,
        searchCandidateById } from '../../util/apiCalls';

jest.mock('../../util/apiCalls');

describe('Committee', () => {

  let wrapper;
  let mockSetPacContributions = jest.fn();
  let mockSetCurrentCommitteeId= jest.fn().mockImplementation(() => {
    wrapper.instance().committee_id = "XYZ";
    wrapper.update();
  });
  let mockSetIndividualContributions = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Committee 
      idFromMatch={"XYZ"} 
      candidate={{name: 'John Hickenlooper'}}
      committee_id={"ABC123"}
      setPacContributions={mockSetPacContributions}
      setCurrentCommitteeId={mockSetCurrentCommitteeId}
      setIndividualContributions={mockSetIndividualContributions}
      setCurrentCommitteeId = {mockSetCurrentCommitteeId}
      />)
  })

  it('should match the snapshot with correct data passed in', () => {
      expect(wrapper).toMatchSnapshot();
    })
 


  describe('componentDidMount', () => {

    beforeEach(() => {
      searchCommitteeById.mockImplementation(() => {
        return Promise.resolve(
          [ {name: 'J', candidateIds: ["ABC123"]}, 
            {name: 'K', candidateIds: ["DEF123"]}, 
            {name: 'L', candidateIds: ["GHI123"]}])
      })

      searchCandidateById.mockImplementation(() => {
        return Promise.resolve([{ name: 'Joe'}] )
      })

      fetchIndividualContributions.mockImplementation(() => {
        return Promise.resolve([{amount: 5}, {amount: 7}])
      })

      fetchPACContributions.mockImplementation(() => {
        return Promise.resolve([{amount: 5}, {amount: 1}])
      })
    })
  
    it('should fire setCurrentCommitteeId if committee_id is null with the id from the url', async () => {
      delete global.window.location;
      global.window.location = { pathname: '/blah/123/committee/ABC123' }
      
      wrapper = shallow(<Committee 
        idFromMatch={undefined} 
        candidate={{name: 'John Hickenlooper'}}
        committee_id={null}
        setPacContributions={mockSetPacContributions}
        setCurrentCommitteeId={mockSetCurrentCommitteeId}
        setIndividualContributions={mockSetIndividualContributions}
        />)

        await wrapper.update()
      expect(mockSetCurrentCommitteeId).toHaveBeenCalledWith('ABC123')
    })
  })

})