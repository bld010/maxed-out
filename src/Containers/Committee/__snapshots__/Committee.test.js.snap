// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Committee should match the snapshot with correct data passed in 1`] = `
<section
  className="Committee"
>
  <div
    className="name-and-info"
  />
  <div
    className="contributions-container"
  >
    <h2>
      Loading
    </h2>
  </div>
</section>
`;
